File: csharp/utils/FileUtils.cs
public class FileUtils
{
    public static List<string> GetInputData(string date)
    {
        if (string.IsNullOrEmpty(date))
            throw new ArgumentNullException(nameof(date));

        var splited = date.Split('-');
        if (!splited.Any())
            throw new ArgumentException("Invalid date format", nameof(date));

        return ReadFileLineByLine(Path.Combine("files", splited.First(), date));
    }

    public static List<string> ReadFileLineByLine(string filepath)
    {
        try
        {
            return File.ReadLines(filepath).ToList();
        }
        catch (FileNotFoundException ex)
        {
            throw new FileNotFoundException($"Input file not found: {filepath}", ex);
        }
        catch (IOException ex)
        {
            throw new IOException($"Error reading file: {filepath}", ex);
        }
    }
}

File: typescript/src/shared/file-utils.ts
import fs from 'fs'
import path from 'path'

const getInputData = (date: string): string[] => {
  if (!date) {
    throw new Error('date parameter cannot be null or empty')
  }

  const splited = date.split('-')
  if (!splited.length) {
    throw new Error('Invalid date format')
  }

  return readFileLineByLine(path.join('files', splited[0], date))
}

const readFileLineByLine = (filepath: string): string[] => {
  try {
    return fs
      .readFileSync(filepath, 'utf8')
      .split('\n')
      .map((line) => line.replace('\r', ''))
      .filter((v) => v)
  } catch (error) {
    if ((error as NodeJS.ErrnoException).code === 'ENOENT') {
      throw new Error(`Input file not found: ${filepath}`)
    }
    throw new Error(`Error reading file: ${filepath}`)
  }
}

export { getInputData, readFileLineByLine }
